dandiapi/api/views/asset.py

```diff
@@ -505,6 +512,35 @@ def destroy(self, request, versions__dandiset__pk, versions__version, **kwargs):
 
         return Response(None, status=status.HTTP_204_NO_CONTENT)
 
+    @swagger_auto_schema(manual_parameters=[ASSET_GLOB_PARAM, ASSET_REGEX_PARAM])
+    def list(self, request, *args, **kwargs):
+        queryset = self.filter_queryset(self.get_queryset())
+
+        glob_pattern: str | None = self.request.query_params.get('glob')
+        regex_pattern: str | None = self.request.query_params.get('regex')
```

Comment:
Unless there's a good reason against it, I'd suggest defining a serializer for these params

```python
class AssetListSerializer(serializers.Serializer):
    glob = serialzers.CharField(required=False)
    regex = serialzers.CharField(required=False)
```

and then just using like so

```python
    @swagger_auto_schema(query_serializer=AssetListSerializer)
    def list(self, request, *args, **kwargs):
        serializer = AssetListSerializer(data=request.query_params)
        serializer.is_valid(raise_exception=True)

        queryset = self.filter_queryset(self.get_queryset())
        glob_pattern: str | None = serializer.validated_data['glob']
        regex_pattern: str | None = serializer.validated_data['regex']
```

IMO `manual_parameters` should be used as a last resort, for when the normal mode of data serialization can't be used.

Reply:
Oh nice, I didn't realize this lets you replace `manual_parameters` - there's other places in the code where `manual_parameters` is used, I wonder if similar refactors could be done there

Reply:
Filed #1026 so we don't forget about this